packages/react-router/lib/dom/ssr/routes.tsx

```diff
@@ -471,6 +473,22 @@ export function createClientRoutes(
         };
       }
 
+      if (route.hasClientMiddleware) {
+        dataRoute.unstable_lazyMiddleware = async () => {
+          invariant(route);
+          let clientMiddlewareModule = await import(
+            /* @vite-ignore */
+            /* webpackIgnore: true */
+            route.clientMiddlewareModule || route.module
```

Comment:
💯 This is really clever - it took me a sec to figure out how it was working without the route chunks since we weren't waiting on the route module promise in `dataStrategy` anymore